jps-plugin/src/org/jetbrains/jps/incremental/scala/CachedCompilationService.scala

```diff
@@ -0,0 +1,62 @@
+package org.jetbrains.jps.incremental.scala
+
+import java.io.File
+
+import org.jetbrains.jps.incremental.scala.data.CompilationData
+import org.jetbrains.jps.incremental.messages.BuildMessage.Kind
+
+import scala.util.Try
+
+
+object CachedCompilationService {
```

Comment:
This is extentension point that allows reusing compilations (similar to [hoarder](https://github.com/romanowski/hoarder))